conda/gateways/repodata/jlap/core.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+Small jlap reader.
+"""
+
+from __future__ import annotations
+
+import logging
+from collections import UserList
+from hashlib import blake2b
+from pathlib import Path
+from typing import Iterable, Iterator
+
+log = logging.getLogger(__name__)
+
+
+DIGEST_SIZE = 32  # 160 bits a minimum 'for security' length?
+DEFAULT_IV = b"\0" * DIGEST_SIZE
+
+
+def keyed_hash(data: bytes, key: bytes):
+    """
+    Keyed hash.
+    """
+    return blake2b(data, key=key, digest_size=DIGEST_SIZE)
+
+
+def line_and_pos(lines: Iterable[bytes], pos=0) -> Iterator[tuple[int, bytes]]:
+    """
+    lines: iterator over input split by '\n', with '\n' removed.
+    pos: initial position
+    """
+    for line in lines:
+        yield pos, line
+        pos += len(line) + 1
+
+
+class JLAP(UserList):
```

Comment:
Since we never access the internal `data` attribute here there's no benefit to using `collections.UserList` over `list[tuple[int, str, str]]`:

```suggestion
class JLAP(list[tuple[int, str, str]]):
```

Reply:
I did try extending list earlier but no that doesn't behave as desired. Iirc you wind up covering back to base list by mistake.